device-types/Juniper/QFX5100-24Q-2P.yaml

```diff
@@ -0,0 +1,99 @@
+---
+manufacturer: Juniper
+model: QFX5100-24Q-2P
+slug: juniper-qfx5100-24q-2p
+part_number: QFX5100-24Q-2P
+u_height: 1
+is_full_depth: true
+airflow: front-to-rear
+weight:
+  - value: 22.0
+    unit: lb
+comments: |
+  [Juniper QFX5100 Data Sheet](https://www.juniper.net/content/dam/www/assets/datasheets/us/en/switches/qfx5100-ethernet-switch-datasheet.pdf)
+
+  [Juniper QFX5100 Chassis Data](https://www.juniper.net/documentation/us/en/hardware/qfx5100/topics/topic-map/qfx5100-port-panels.html)
+
+interfaces:
+  - name: et-0/0/0
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/1
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/2
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/3
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/4
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/5
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/6
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/7
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/8
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/9
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/10
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/11
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/12
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/13
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/14
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/15
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/16
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/17
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/18
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/19
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/20
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/21
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/22
+    type: 40gbase-x-qsfpp
+  - name: et-0/0/23
+    type: 40gbase-x-qsfpp
+  - name: em0
+    type: 1000base-t
+    mgmt_only: true
+  - name: em1
+    type: 1000base-x-sfp
+    mgmt_only: true
+power-ports:
+  - name: PSU0
+    type: iec-60320-c14
+    maximum_draw: 650
+  - name: PSU1
+    type: iec-60320-c14
+    maximum_draw: 650
+module-bays:
+  - name: Fan Tray 0
+  - name: Fan Tray 1
+  - name: Fan Tray 2
+  - name: Fan Tray 3
+  - name: Fan Tray 4
+  - name: PIC1
+    label: Expansion module QFX-EM-4Q
```

Comment:
These are not the only modules that can fit in there. Please be more generic in your labels.
![image](https://user-images.githubusercontent.com/1564902/221010702-53356828-2c33-4707-a29c-3c03e0e0bbcd.png)